core/src/main/java/io/grpc/HandlerRegistry.java

```diff
@@ -42,6 +44,16 @@
 public abstract class HandlerRegistry {
 
   /**
+   * Returns the {@link ServerServiceDefinition}s provided by the registry.
+   *
+   * @throws UnsupportedOperationException if not supported by implementation
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2222")
+  public List<ServerServiceDefinition> getServices() {
```

Comment:
List is the right one.  Looking further down, this is going to have allocate a list anyways from the MutableHandlerRegistry  map.